monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/commons/auth/AuthenticationFilter.java

```diff
@@ -109,4 +142,37 @@ private void abortWithUnauthorized(ContainerRequestContext requestContext) {
     private boolean inValidToken(String token) {
         return TokenCache.getInstance().getToken(token) == null;
     }
+
+    private boolean inValidSelfContainedToken(SSOConfig config, String token) {
+
+        if (config == null) {
+            return true;
+        }
+        JWT idTokenJWT = null;
+        try {
+            idTokenJWT = JWTParser.parse(token);
+            IDTokenValidator validator = new IDTokenValidator(config.getOidcAgentConfig(), idTokenJWT);
+            validator.validate(null);
+            return false;
+        } catch (ParseException | SSOAgentServerException e) {
+            if (logger.isDebugEnabled()) {
+                logger.error("Error validating the access token", e);
+            }
+        }
+        return true;
+    }
+
+    private boolean isUserInAdminGroup(JsonElement tokenPayload, SSOConfig config) {
+
+        if (config == null) {
+            return false;
+        }
+        JsonArray groupElement = tokenPayload.getAsJsonObject().getAsJsonArray(config.getAdminGroupAttribute());
```

Comment:
Wouldn't it be more clear if we can convert the token JSON into a Java Object using Gson ( we can use the `gson.fromJson` function). Then we can use that object to get the values in the token other than parsing it every time we need. It would also give future developers a clear view of the token structure also.

Reply:
Since we use dynamic keys to get the elements from the token, we may need to use a map inside our Java class which will have a similar outcome to this approach.

Reply:
Then this way is better I guess